src/gate_detect/src/Gate.cpp

```diff
@@ -0,0 +1,189 @@
+#include "Gate.h"
+
+Gate::Gate() {
+    lowVertThresh = 50; // Sets the maximum deviation from start to finish that
+                        // will constitute a vertical line
+    lowHorThresh = 100; // Sets the maximum deviation from start to finish that
+                        // will constitute a horizontal line
+    cannyLow = 50; // Sets the low threshold for the canny filter, adjust if dst
+                   // is not displaying all lines
+    cannyHigh = 200; // Sets the high threshold for the canny filter, adjust if
+                     // dst not displaying all lines
+    poleMax =
+    200; // Sets the minimum threshold for two lines to be called a pole
+    m                = -0.002673; // Readjust when calibrated
+    b                = 5.134;     // Distance from gate = m * x + b
+    camera_res_width = 1920;
+}
+
+std::vector<float> Gate::initialize(const cv::Mat matin) {
+    cv::Canny(matin, dst, cannyLow, cannyHigh, 3);
+
+    cv::HoughLinesP(dst, lines, 1, CV_PI / 180, 60, 50, 50);
+
+    if (lines.empty()) return std::vector<float>(9, 0);
+
+    _vertLines = filterVertLines(lines);
+
+    _vertPoles = findVertPoles(_vertLines);
+
+    _horLines = filterHorLines(lines);
+
+    _horPoles = findHorPoles(_horLines);
+
+    poleVector = gateVector(_vertPoles, _horPoles);
+
+    return poleVector;
+}
+
+bool Gate::checkMat() {
+    return (!dst.empty());
+}
+
+std::vector<int> Gate::filterVertLines(std::vector<cv::Vec4i> allLines) {
+    std::vector<int> vertLines;
+    vertLines.clear();
+
+    for (int i = 0; i < (allLines.size() - 1); i++) {
+        cv::Vec4i newLine = allLines[i];
+        if ((abs(newLine[0] - newLine[2]) < lowVertThresh)) {
+            vertLines.push_back((newLine[0] + newLine[2]) / 2);
+        }
+    }
+
+    return vertLines;
+}
+
+std::vector<int> Gate::filterHorLines(std::vector<cv::Vec4i> allLines) {
+    std::vector<int> horLines;
+    horLines.clear();
```

Comment:
shouldn't need to clear this vector as it was just created.

Reply:
Removed